backend/module/chaos/serverexperimentation/xds/ecds.go

```
@@ -0,0 +1,255 @@
+		switch httpFaultConfig.GetFaultTargeting().GetEnforcer().(type) {
+		case *serverexperimentation.FaultTargeting_DownstreamEnforcing:
+			// Abort External Fault
+			isExternalFault = true
+
+		case *serverexperimentation.FaultTargeting_UpstreamEnforcing:
+			// Abort Internal Fault
+			isExternalFault = false
+
+		default:
+			// If error, return default abort and default delay configs
+			return false, abort, delay, fmt.Errorf("unknown enforcer %v", httpFaultConfig)
+		}
+
+		abort = &gcpFilterFault.FaultAbort{
+			ErrorType: &gcpFilterFault.FaultAbort_HttpStatus{
+				HttpStatus: httpFaultConfig.GetAbortFault().GetAbortStatus().GetHttpStatusCode(),
+			},
+			Percentage: &gcpType.FractionalPercent{
+				Numerator:   httpFaultConfig.GetAbortFault().GetPercentage().GetPercentage(),
+				Denominator: gcpType.FractionalPercent_HUNDRED,
+			},
+		}
+	case *serverexperimentation.HTTPFaultConfig_LatencyFault:
+		switch httpFaultConfig.GetFaultTargeting().GetEnforcer().(type) {
+		case *serverexperimentation.FaultTargeting_DownstreamEnforcing:
+			// Latency External Fault
+			isExternalFault = true
+
+		case *serverexperimentation.FaultTargeting_UpstreamEnforcing:
+			// Latency Internal Fault for all downstream services
+			isExternalFault = false
+
+		default:
+			// If error, return default abort and default delay configs
+			return false, abort, delay, fmt.Errorf("unknown enforcer %v", httpFaultConfig)
+		}
+
+		delay = &gcpFilterCommon.FaultDelay{
+			FaultDelaySecifier: &gcpFilterCommon.FaultDelay_FixedDelay{
+				FixedDelay: &duration.Duration{
+					Seconds: int64(httpFaultConfig.GetLatencyFault().GetLatencyDuration().GetFixedDurationMs()),
+				},
+			},
+			Percentage: &gcpType.FractionalPercent{
+				Numerator:   httpFaultConfig.GetLatencyFault().GetPercentage().GetPercentage(),
+				Denominator: gcpType.FractionalPercent_HUNDRED,
+			},
+		}
+	default:
+		return false, abort, delay, fmt.Errorf("unknown fault type %v", httpFaultConfig)
```

I would return nil instead of a default abort/delay in the error cases

---

if there is an error, would it not be better to send default configs (which will revert the fault)?